### PR TITLE
Initial work for sdg on ocp

### PR DESCRIPTION
**Pull request by MichaelClifford**
Initial Containerfile, deployment.yaml and python script to run SDG on OCP. 

Admittedly this is a fairly rigid example, where I've updated the `Taxonomy/` repo  as part of the container build so that it will always start with a git diff for the SDG to use (SDG seems to require a git diff to pick out what data will be used for generation). Currently there is no real way for a user to update the taxonomy with this code, hopefully we can address that in a later PR when we determine how a user will actually interact with this whole pipeline. 

To run this code on OCP you must: 
* set OPENAI_API_KEY to your Authorization Token for your route 
* OPENAI_BASE_URL to your route address
* OPENAI_MODEL to the model name in the vllm model server.

### CODE REVIEW

**Comment by cooktheryan** [comment]
@tumido

File: eval/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      containers:
+        - name: container
+          image: quay.io/michaelclifford/test-sdg:0.0.0
+          resources:
+          command:
+            - /bin/bash
+            - '-c'
+            - '--'
+          args:
+            - while true; do sleep 300; done
```
Comment (Shreyanand) [inline eval/deployment.yaml:25]: should we add `python3.11 eval/run.py` here

**Review by Shreyanand — COMMENTED** [review]
Looks good overall 👍

**Review by hemajv — COMMENTED** [review]
Any reason these files are under the `/eval` folder and not the `/SDG` folder?

Else, everything else looks good to me 👍

File: eval/deployment.yaml (hunk 1 above)
Comment (MichaelClifford) [reply eval/deployment.yaml:25]: yeah, sure.

**Comment by MichaelClifford** [comment]
@hemajv Good catch, lol. definitely in the wrong directory 😄

File: SDG/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      containers:
+        - name: container
+          image: quay.io/michaelclifford/test-sdg:0.0.0
+          resources:
+          command:
+            - /bin/bash
+            - '-c'
+            - '--'
+          args:
+            - while true; do sleep 300; done
```
Comment (JudeNiroshan) [inline SDG/deployment.yaml:25]:
```suggestion
            - python3.11 run.py; while true; do sleep 300; done
```

File: SDG/run.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+if not os.getenv("OPENAI_API_KEY"):
+    os.environ["OPENAI_API_KEY"] = "Not_Needed"
+if not os.getenv("OPENAI_BASE_URL"):
+    os.environ["OPENAI_BASE_URL"]  = "http://10.88.0.1:8001/v1"
+if not os.getenv("OPENAI_MODEL"):
+     os.environ["OPENAI_MODEL"] = "Not_Needed"
+
+client = openai.OpenAI()
+openai.default_headers = {"Authorization": f'Bearer:{os.environ["OPENAI_API_KEY"]}'}
```
Comment (JudeNiroshan) [inline SDG/run.py:13]: Should we be more specific about this? something like `MODEL_SERVER_TOKEN`

**Review by JudeNiroshan — APPROVED** [review]

File: SDG/run.py (hunk 3 above)
Comment (hemajv) [reply SDG/run.py:13]: +1, I think that would provide more clarity

**Comment by tumido** [comment]
Do we need this when we have functional #2 ?🙂

**Comment by MichaelClifford** [comment]
@tumido yes, lets close this PR in favor of #2. It looks like all the relevant bits have been incorporated over there 😄

**Pull request closed without merging**